### ISSUE TITLE
Project Coding Standards being ignored

### ISSUE THREAD

**Issue by eggsurplus** [salesagility/SuiteCRM#6832]
There is a nice Coding Standards doc for SuiteCRM here: 

[https://docs.suitecrm.com/community/contributing-code/coding-standards/](https://docs.suitecrm.com/community/contributing-code/coding-standards/)

Noticing on each release a battle where files change from the PSR-2 standard to whatever a dev feels like and vice versa. For example, in 7.11.0 there are many places where indentation has changed from 4 spaces to indents, spacing after conditional statements removed, curly braces moved to same line as conditional, braces being completely removed when it is a single line (the savagery!), etc. 

See include/generic/SugarWidgets/SugarWidgetFieldenum.php for an example. Here is a file comparison. The 4 space indentation has changed to tabs:

[https://www.dropbox.com/s/og1bcjoi17ten2z/Screenshot%202019-01-24%2021.05.38.png?dl=0](https://www.dropbox.com/s/og1bcjoi17ten2z/Screenshot%202019-01-24%2021.05.38.png?dl=0)

Seeing plenty of other examples in the latest release and a trend of this in other recent releases. It would be great to get back to the defined Coding Standards.

**Comment by Dillon-Brown** [comment]
I would disagree that the project coding standards are being ignored. I certainly wouldn't approve any PR that changes spaces to tabs!

The issue here is actually due to the fact we have multiple branches (7.8.x 7.10.x 7.11.x) which are at different levels of PSR-2 compliance. If a developer makes a change to a function it's expected that they would fix PSR-2 issues in that function but not the entire file.

Here's some of the work I've been doing for PSR-2:

https://github.com/salesagility/SuiteCRM/pull/6090
https://github.com/salesagility/SuiteCRM/pull/6083
https://github.com/salesagility/SuiteCRM/pull/6084
https://github.com/salesagility/SuiteCRM/pull/6085
https://github.com/salesagility/SuiteCRM/pull/6086
https://github.com/salesagility/SuiteCRM/pull/6087
https://github.com/salesagility/SuiteCRM/pull/6088
https://github.com/salesagility/SuiteCRM/pull/6089
https://github.com/salesagility/SuiteCRM/pull/6091
https://github.com/salesagility/SuiteCRM/pull/6093
https://github.com/salesagility/SuiteCRM/pull/6094
https://github.com/salesagility/SuiteCRM/pull/6095
https://github.com/salesagility/SuiteCRM/pull/6359

However with the recent merging of develop to hotfix these changes will need re-done. I would definitely say the trend is positive though. The hotfix-7.10.x branch should be almost if not fully PSR-2 compliant and the hotfix branch will quickly be brought up to speed.

**Comment by Dillon-Brown** [comment]
Just to also mention, I've created a Robo script here: #6186 which will allow anyone from the community that wants to help out run PSR-2 fixers. 

Thanks for raising this, hopefully we can get #6836 merged in soon and be fully PSR-2 compliant!

**Comment by eggsurplus** [comment]
Thanks Dillon for the additional insight of how there are multiple branches in parallel. This would explain the larger number of changes of late that appears to be going away from the standards. Hopefully that's the case and over the next several branches those issues go away. 

I do acknowledge that there have been improvements over the past couple of years which is why it was alarming to see such a drastic jump backwards in 7.11.0. In the example I gave before with SugarWidgetFieldenum, the displayListPlain function was cleaned up in 7.10.9. It was odd to then see it revert back much later in the 7.11.0 release. What is the process to get improvements in 7.10 that happened after the initial 7.11 branch back into 7.11?

Similar note, would it be helpful to add a hook to the commit process that automatically checks for and enforces coding standards?

**Comment by Dillon-Brown** [comment]
@eggsurplus The current process is to create separate pull requests for each branch when making a fix as trying to merge the branches together tends to cause more problems than it solves (These PSR-2 fixes are around 100k line changes!). I've got a PR to fix all the coding standard issues on 7.11 which we'll look at merging in for the next release. 

I completely agree that we should be checking the coding standards on every PR automatically. I've implemented [Codacy](https://app.codacy.com) for this I still need to look at creating configuration files as it's currently flagging a few things we can safely ignore.
